tools/cbmc/proofs/proof_specific_makefile.py

```diff
@@ -376,16 +425,32 @@ def set_up_logging(args):
     else:
         logging.basicConfig(format=fmt, level=logging.WARNING)
 
+
 def wrap(string):
     return re.sub(r"\s+", " ", re.sub("\n", " ", string))
 
+
 def main():
     args = get_args()
     set_up_logging(args)
 
     for root, _, fyles in os.walk("."):
         if "Makefile.json" in fyles:
-            dump_makefile(root, args.system)
+            makefile_lines = get_makefile(root, args.system)
+
+            # Deal with the case of a harness being nested several levels under
+            # the top-level proof directory, where the common Makefile lives
+            common_dir_path = "..%s" % _platform_choices[args.system]["path-sep"]
+            common_dir_path = common_dir_path * len(root.split(os.path.sep)[1:])
+
+            with open(os.path.join(root, "Makefile"), "w") as handle:
+                handle.write("""\
+{contents}
+
+{include} {common_dir_path}Makefile.common""".format(
```

Comment:
Why does this makefile refer to makefile.common?  I thought the whole point was to generate just a single makefile.